src/components/cylc/gantt/GanttChart.vue

```diff
@@ -0,0 +1,212 @@
+  />
+</template>
+
+<script>
+import VueApexCharts from 'vue3-apexcharts'
+import {
+  mdiDownload,
+  mdiSortReverseVariant,
+  mdiSortVariant,
+} from '@mdi/js'
+
+export default {
+  name: 'GanttChart',
+
+  components: {
+    VueApexCharts,
+  },
+
+  props: {
+    jobs: {
+      type: Array,
+      required: true,
+    },
+    timingOption: {
+      type: String,
+      required: true,
+    },
+    itemsPerPage: {
+      type: Number,
+      default: 20,
+    },
+    animate: {
+      type: Boolean,
+      default: true,
+    },
+    /** Where to teleport the sorting input (or don't render if null) */
+    sortInputTeleportTarget: {
+      type: String,
+      default: null,
+    },
+  },
+
+  data () {
+    return {
+      page: 1,
+      sortBy: 'name',
+      sortDesc: false,
+    }
+  },
+
+  computed: {
+    series () {
+      const data = []
+      if (this.jobs.length !== 0) {
+        const colours = [
+          '#008FFB',
+          '#00E396',
+          '#775DD0',
+          '#FEB019',
+          '#FF4560',
+        ]
+        const jobColours = {}
+        let jobNameList = new Set([])
+
+        for (let i = 0; i < this.jobs.length; i++) {
+          jobNameList.add(this.jobs[i].name)
+        }
+        jobNameList = Array.from(jobNameList)
+
+        for (let i = 0; i < jobNameList.length; i++) {
+          jobColours[jobNameList[i]] = colours[i % colours.length]
+        }
+
+        const startIndex = 0
+        const endIndex = this.jobs.length
+
+        if (this.timingOption === 'total') {
+          for (let i = startIndex; i < endIndex; i++) {
+            data.push({
+              x: this.jobs[i].name,
+              y: [
+                new Date(this.jobs[i].submittedTime).getTime(),
+                new Date(this.jobs[i].finishedTime).getTime(),
+              ],
+              fillColor: jobColours[this.jobs[i].name],
+            })
+          }
+        } else if (this.timingOption === 'run') {
+          for (let i = startIndex; i < endIndex; i++) {
+            data.push({
+              x: this.jobs[i].name,
+              y: [
+                new Date(this.jobs[i].startedTime).getTime(),
+                new Date(this.jobs[i].finishedTime).getTime(),
+              ],
+              fillColor: jobColours[this.jobs[i].name],
+            })
+          }
+        } else if (this.timingOption === 'queue') {
+          for (let i = startIndex; i < endIndex; i++) {
+            data.push({
+              x: this.jobs[i].name,
+              y: [
+                new Date(this.jobs[i].submittedTime).getTime(),
+                new Date(this.jobs[i].startedTime).getTime(),
+              ],
+              fillColor: jobColours[this.jobs[i].name],
+            })
+          }
+        }
+      }
+      return [{ data }]
+    },
+
+    chartOptions () {
+      return {
+        chart: {
+          animations: {
+            enabled: this.$store.state.app.reducedAnimation ? false : this.animate,
```

Comment:
This has changed to use a composable rather than the store

```diff
diff --git a/src/components/cylc/gantt/GanttChart.vue b/src/components/cylc/gantt/GanttChart.vue
index 9b5ce490..6341db89 100644
--- a/src/components/cylc/gantt/GanttChart.vue
+++ b/src/components/cylc/gantt/GanttChart.vue
@@ -33,4 +33,5 @@ import {
   mdiSortVariant,
 } from '@mdi/js'
+import { useReducedAnimation } from '@/composables/localStorage'
 
 export default {
@@ -141,5 +142,5 @@ export default {
         chart: {
           animations: {
-            enabled: this.$store.state.app.reducedAnimation ? false : this.animate,
+            enabled: useReducedAnimation().value ? false : this.animate,
             easing: 'easeinout',
             speed: 300,
```